src/sync/arc.rs

```diff
@@ -25,8 +25,30 @@ impl<T> Arc<T> {
     }
 
     /// Returns the inner value, if the `Arc` has exactly one strong reference.
-    pub fn try_unwrap(_this: Arc<T>) -> Result<T, Arc<T>> {
-        unimplemented!();
+    pub fn try_unwrap(this: Arc<T>) -> Result<T, Arc<T>> {
+        if this.obj.get_mut(location!()) {
+            assert_eq!(1, std::sync::Arc::strong_count(&this.value));
+
+            // work around our inability to destruct the object normally,
+            // because of the `Drop` presense.
+            this.obj.ref_dec(location!());
+            this.unregister();
+
+            let arc_value = unsafe {
+                let _arc_obj = ptr::read(&this.obj);
+                let arc_value = ptr::read(&this.value);
+
+                mem::forget(this);
+
+                arc_value
+            };
```

Comment:
a potential safe alternative to this is to wrap the loom `Arc`'s value field in an `Option` so that we can `take` it out without dropping the struct...but, this would mean every other time the `Arc`'s value is accessed would have to `unwrap()` it, which might have performance implications and make tests slower...this might be worth the unsafe code to avoid that? 

it would be nice to add a comment explaining why this is safe, though.

Reply:
I don't have a strong preference. Current approach is borrowed from libstd. Added a comment now.

Reply:
yeah, i think doing it similarly to the way it's implemented in `std` is fine. thanks for the comment!